src/pages/Transfer/CrossChainTransferForm/CrossChainTransferForm.tsx

```diff
@@ -35,125 +41,253 @@ import {
+
+    const destinationChains = getDestinationChains(chain);
+
+    setDestinationChains(destinationChains);
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_FIELD, destinationChains[0].id);
+  };
+
+  const handleDestinationChainChange = async (chain: ChainName, name: string) => {
+    if (!accountId) return;
+
+    form.setFieldValue(name, chain);
+
+    const tokens = await getAvailableTokens(
+      form.values[CROSS_CHAIN_TRANSFER_FROM_FIELD] as ChainName,
+      chain,
+      accountId.toString(),
+      form.values[CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD] as string
+    );
+
+    if (!tokens) return;
 
-    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD, accountId.toString());
-    form.validateField(CROSS_CHAIN_TRANSFER_AMOUNT_FIELD);
+    setTransferableTokens(tokens);
+    setCurrentToken(tokens[0]);
+  };
+
+  useEffect(() => {
+    if (!transferableTokens.length) return;
+    const defaultToken = transferableTokens[0];
+
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TOKEN_FIELD, defaultToken.value);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [accountId]);
+  }, [transferableTokens]);
 
-  const transferMonetaryAmount = newSafeMonetaryAmount(
-    form.values[CROSS_CHAIN_TRANSFER_AMOUNT_FIELD] || 0,
-    GOVERNANCE_TOKEN,
-    true
-  );
+  const handleTickerChange = (ticker: string, name: string) => {
+    form.setFieldValue(name, ticker);
+    setCurrentToken(transferableTokens.find((token) => token.value === ticker));
+  };
+
+  const handleDestinationAccountChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD, e.target.value);
+  };
+
+  const transferMonetaryAmount = currentToken
+    ? newSafeMonetaryAmount(
+        form.values[CROSS_CHAIN_TRANSFER_AMOUNT_FIELD] || 0,
+        getCurrencyFromTicker(currentToken.value),
+        true
+      )
+    : 0;
+
+  const valueUSD = transferMonetaryAmount
+    ? convertMonetaryAmountToValueInUSD(
+        transferMonetaryAmount,
+        getTokenPrice(prices, currentToken?.value as string)?.usd
+      )
+    : 0;
+
+  const isCTADisabled = isFormDisabled(form) || form.values[CROSS_CHAIN_TRANSFER_AMOUNT_FIELD] === '';
+
+  useEffect(() => {
+    if (!originatingChains?.length) return;
 
-  const valueUSD =
-    convertMonetaryAmountToValueInUSD(transferMonetaryAmount, getTokenPrice(prices, GOVERNANCE_TOKEN.ticker)?.usd) ?? 0;
+    // This prevents a render loop caused by setFieldValue
+    if (form.values[CROSS_CHAIN_TRANSFER_FROM_FIELD]) return;
 
-  const isCTADisabled = isFormDisabled(form);
+    const destinationChains = getDestinationChains(originatingChains[0].id);
+
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_FROM_FIELD, originatingChains[0].id);
+    form.setFieldValue(CROSS_CHAIN_TRANSFER_TO_FIELD, destinationChains[0].id);
+
+    setDestinationChains(destinationChains);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [originatingChains]);
+
+  useEffect(() => {
```

Comment:
Yeah—this will need to be changed to handle refetching the token data after a transaction as well, which would be a third instance of very similar code. Let's leave it as it is in this PR, and I'll move it to either a function in the component or a method in the hook, and get rid of the code duplication. (Refetching balances will be handled in a separate PR).